zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -50,14 +50,10 @@ where
         .or_else(|| params.activation_height(NetworkUpgrade::Sapling))
         .ok_or(Error::SaplingNotActive)?;
 
-    let sapling_outputs = decrypt_transaction(params, height, tx, &ufvks);
-
-    if !(sapling_outputs.is_empty() && tx.transparent_bundle().iter().all(|b| b.vout.is_empty())) {
```

Comment:
This change means that the transaction passed to `decrypt_and_store_transaction` is always saved to the wallet, instead of only when it is of interest to us. I think this is within the documented behaviour of this method, so it's fine.